nni/tools/package_utils/config_manager.py

```diff
@@ -9,24 +11,26 @@
 ]
 
 from collections import defaultdict
-from typing import List, Optional
 
 import yaml
 
 from nni.runtime.config import get_builtin_config_file, get_config_file
 from .common import AlgoMeta
 
-def get_algo_meta(name: AlgoMeta) -> Optional[AlgoMeta]:
+def get_algo_meta(name: str) -> AlgoMeta | None:
     """
     Get meta information of a built-in or registered algorithm.
     Return None if not found.
     """
+    name = name.lower()
     for algo in get_all_algo_meta():
-        if algo.name == name:
+        if algo.name.lower() == name:
+            return algo
+        if algo.alias is not None and algo.alias.lower() == name:
             return algo
     return None
 
-def get_all_algo_meta() -> List[AlgoMeta]:
+def get_all_algo_meta() -> list[AlgoMeta]:
```

Comment:
when should we use `List` or never use it?

Reply:
It's just style preference.
I personally prefer `list` so modules maintained by me will always use `list`.